src/ConfigResolver/index.ts

```diff
@@ -0,0 +1,91 @@
+import inclusion from 'inclusion'
+import { pathToFileURL } from 'url'
+import { ReporterContract, Refiner, NamedReporterContract } from '@japa/core'
+import { Config, NormalizedConfig } from '../Contracts'
+
+/**
+ * Wrap the value inside an array if it's not one
+ */
+function arrayify<T>(value: T | T[]): T[] {
+  return Array.isArray(value) ? value : [value]
+}
+
+/**
+ * Convert a ReporterContract object to a NamedReporterContract object
+ */
+function buildReporterConfig(reporters: ReporterContract[]): NamedReporterContract[] {
+  return reporters.map((reporter) => {
+    /**
+     * If the reporter wasn't named, then we generate a random name
+     */
+    const reporterName = reporter.name || Math.random().toString(36).substring(5)
+
+    if (typeof reporter === 'function') {
+      return { handler: reporter, name: reporterName }
+    }
+
+    return reporter
+  })
+}
+
+/**
+ * Build the normalized `reporters` configuration
+ */
+function buildReportersConfig(
+  options: Config & { defaultsReporters?: string[] }
+): NormalizedConfig['reporters'] {
+  /**
+   * Handle the new runner.reporters API
+   */
+  if (options.reporters && 'list' in options.reporters) {
+    let defaultsReporters = options.defaultsReporters || options.reporters.defaults || []
```

Comment:
Little confused, what is this `defaultsReporters` property?

Reply:
Forgot about explaining that, sorry

So, our configuration looks like this :
```ts
configure({
   ....processCliArgs(),
   ...{
     reporters: {
       list: [],
       defaults: []
     }
   }
})
```

We merge the result of processCliArgs, in the final configuration. The problem with a simple merge like this is that **it is not deep**

So the `defaultsReporters` property comes from the `processCliArgs` return. If we were returning a `reporters.defaults` instead, and it would also be defined in the user's configuration, then it would be totally overwritten.

That's why I introduced this new property. I'm clearly not a fan of it, but it was the only solution I could see.

In the end, what I propose is to hide this "processCliArgs" from the user and let the configure function handle the CLI arguments itself.

This way, we could do deep merge without issues. And it would also simplify the API, which I find a little too low level. We could imagine something like this :

```ts
configure({
   processCliArgs: false // defaults to true
   reporters: {
     list: [],
     defaults: []
   }
})
```

Reply:
Ohhh I see. Can we rename it to `cliReporters`?